py4cast/writing_outputs.py

```diff
@@ -0,0 +1,255 @@
+import datetime as dt
+from copy import deepcopy
+from pathlib import Path
+from typing import Any, Tuple
+
+import numpy as np
+import xarray as xr
+from cfgrib import xarray_to_grib as xtg
+
+from py4cast.datasets.base import DatasetABC, NamedTensor
+from py4cast.forcingutils import compute_hours_of_day
+
+
+def saveNamedTensorToGrib(
+    pred: NamedTensor,
+    ds: DatasetABC,
+    sample: Any,
+    date: dt.datetime,
+    saving_settings: dict,
```

Comment:
THe problem of having a dict **saving_settings** as a function parameter is that you MUST KNOW the keys inside and their types to work with/use the function. As it is the docstring does not list the exact keys but some description in english so one must  do the bijection reading the code. This might require reading the while code of the function. 

Some alternatives:
* direct keyword arguments to the function (if not too many)
* a settings dataclass
* a namedtuple

As it is the proposed templating system might be a bit complex don't you think ? I think such an approach would require at least a sound default working value ?

Reply:
Also with [dataclass json](https://pypi.org/project/dataclasses-json/) you can (de)serialize from/to a json file with Schema validation. (I saw you have a json file for setting up your outputs)

Reply:
> THe problem of having a dict **saving_settings** as a function parameter is that you MUST KNOW the keys inside and their types to work with/use the function. As it is the docstring does not list the exact keys but some description in english so one must do the bijection reading the code. This might require reading the while code of the function.
> 
> Some alternatives:
> 
>     * direct keyword arguments to the function (if not too many)
> 
>     * a settings dataclass
> 
>     * a namedtuple

True, my initial thought was a json schema validation with dataclass.

Reply:
> As it is the proposed templating system might be a bit complex don't you think ? I think such an approach would require at least a sound default working value ?

Reply:
Json dataclass done, reorganized modules, name templating set to minimal working example.